docs/concepts/package-types.md

```diff
@@ -0,0 +1,225 @@
+| `esm-view` | Yes             | Yes             | Yes            | Yes            | No                         | Yes + rewriting external dependencies to CDN | The compiled bundle will use `src/index.tsx` as an entrypoint and [link it](../esm-views/output-package-manifest.md) in the output `package.json`; the entrypoint needs to default export a React component for `modular start`/ [standalone `index.html`](../esm-views/how-to-build.md) to work |
+| `package`  | No              | Yes             | Yes            | Yes            | N/A                        | No                                           | Whatever specified in `main` field of `package.json` is used as entrypoint                                                                                                                                                                                                                       |
+| `view`     | Yes             | Yes             | Yes            | Yes            | N/A                        | No                                           | Whatever specified in `main` field of `package.json` is used as entrypoint; needs to default export React component for `modular start` to work                                                                                                                                                  |
+| `source`   | No              | No              | Yes            | Yes            | N/A                        | N/A - never built                            | N/A - don't need the concept of entrypoint since this package is never built                                                                                                                                                                                                                     |
+| `template` | No              | No              | Yes            | Yes            | Depends on the target type | N/A - never built                            | N/A - don't need the concept of entrypoint since this is a special type of package used by `modular add` which is never built directly                                                                                                                                                           |
+
+## App
+
+Modular `app`s are Typescript and React based Single Page Applications (SPA)
+built with [Webpack v5](https://webpack.js.org/) (by default) or
+[esbuild](https://esbuild.github.io/) (by turning on `useModularEsbuild` in
+[the configuration](../configuration.md)).
+
+Apps are generated by `modular add` using the
+[`modular-template-app`](https://github.com/jpmorganchase/modular/tree/main/packages/modular-template-app)
+[template](../concepts/templates.md) and need at least a `public/index.html`
+file and an entrypoint located in `src/index.tsx`, that typically `render`s a
+React component to the DOM, to work.
+
+Apps support pretty much all the functionalities from
```

Comment:
Just a thought on this.

Not sure if this is covered by other docs but it might be good to be a bit more transparent about how modular-scripts relates to CRA. Also, this isn't specific to `apps` but obviously underpins them (and the other types), so not sure where it would go.

I'm thinking we might want to say something like:

> Modular is based on Create React App v5 and maintains a high level of compatibility, but is an independent project that makes it's own choices (and has it's own features) that occasional diverge from Create React App ... _details_

I say this because 
- It's a good starting place if the user is familiar with CRA, and they can refer to that project and their existing knowledge
- We make it clear that it's not literally CRA, and they should refer to these docs